substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -1146,24 +1603,46 @@ private List<DebugLocalInfo> createParamInfo(ResolvedJavaMethod method, int line
             String name = (local != null ? local.getName() : "__" + i);
             ResolvedJavaType paramType = (ResolvedJavaType) signature.getParameterType(i, null);
             JavaKind kind = paramType.getJavaKind();
-            JavaKind storageKind = isPseudoObjectType(paramType, ownerType) ? JavaKind.Long : kind;
+            JavaKind storageKind = isForeignWordType(paramType, ownerType) ? JavaKind.Long : kind;
             paramInfos.add(new NativeImageDebugLocalInfo(name, storageKind, paramType, slot, line));
             slot += kind.getSlotCount();
         }
         return paramInfos;
     }
 
     /**
-     * Identify a pseudo-object Java type which is used only to model a memory word, pointer or
-     * foreign opaque type.
-     * 
+     * Identify a Java type which is being used to model a foreign memory word or pointer type.
+     *
      * @param type the type to be tested
      * @param accessingType another type relative to which the first type may need to be resolved
-     * @return true if the type is a pseudo object type
+     * @return true if the type models a foreign memory word or pointer type
+     */
+    private boolean isForeignWordType(JavaType type, ResolvedJavaType accessingType) {
+        assert accessingType instanceof HostedType : "must be!";
```

Comment:
I tend to avoid assertions like that because they don't provide much benefit over the `ClassCastException` that would be thrown anyways.

Reply:
Sure, removed.